lib/rules/no-implicit-coercion.js

```diff
@@ -317,6 +325,14 @@ module.exports = {
                     report(node, recommendation, true);
                 }
 
+                // foo - 0
+                operatorAllowed = options.allow.includes("-");
+                if (!operatorAllowed && options.number && node.operator === "-" && isNumeric(node.right) && node.right.value === 0 && !isNumeric(node.left)) {
```

Comment:
```suggestion
                if (!operatorAllowed && options.number && node.operator === "-" && node.right.type === "Literal" && node.right.value === 0 && !isNumeric(node.left)) {
```

It's sufficient to check the type of `node.right`, since we can be sure that the value will be strictly `0` after the next comparison.